asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -0,0 +1,223 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union  # noqa: F401
+from uuid import UUID
+
+from asapdiscovery.data.openeye import (
+    _get_SD_data_to_object,
+    _set_SD_data_repr,
+    get_SD_data,
+    oechem,
+    oemol_to_inchi,
+    oemol_to_inchikey,
+    oemol_to_sdf_string,
+    oemol_to_smiles,
+    print_SD_Data,
+    sdf_string_to_oemol,
+    set_SD_data,
+    smiles_to_oemol,
+)
+from pydantic import UUID4, Field
+
+from .experimental import ExperimentalCompoundData
+from .schema_base import (
+    DataModelAbstractBase,
+    DataStorageType,
+    read_file_directly,
+    write_file_directly,
+)
+
+
+class InvalidLigandError(ValueError):
+    ...
+
+
+# Ligand Schema
+
+
+class LigandIdentifiers(DataModelAbstractBase):
+    """
+    Identifiers for a Ligand
+    """
+
+    moonshot_compound_id: Optional[str] = Field(
```

Comment:
Is there some structure to these ids? if so it might be good to add some validation in the form of regex to make sure invalid ids don't sneak through.

Reply:
I'm sure @kaminow can cook some nice regex up for us!

We used to use REGEX to parse all the IDs out of the file names so this definitely seems possible!

Reply:
Raised an issue #344